apis/kms/v1beta1/kmskeyring.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/clients/generated/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	KMSKeyRingGVK = schema.GroupVersionKind{
+		Group:   SchemeGroupVersion.Group,
+		Version: SchemeGroupVersion.Version,
+		Kind:    "KMSKeyRing",
+	}
+)
+
+// +kcc:proto=google.cloud.kms.v1.KeyRing
+type KMSKeyRingSpec struct {
+	// /* Immutable. The Project that this resource belongs to. */
+	// ProjectRef refs.ProjectRef `json:"projectRef"`
+
+	/* Immutable. Optional. The name of the resource. Used for creation and acquisition. When unset, the value of `metadata.name` is used as the default. */
+	// +optional
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Immutable. The location for the KeyRing.
+	//                  A full list of valid locations can be found by running 'gcloud kms locations list'.
+	// +required
+	Location *string `json:"location,omitempty"`
+}
+
+// +kcc:proto=google.cloud.kms.v1.KeyRing
+type KMSKeyRingStatus struct {
+	/* Conditions represent the latest available observations of the
+	   KMSKeyRing's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	/* ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource. */
+	// +optional
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	//   // Output only. The resource name for the
+	// //  [KeyRing][google.cloud.kms.v1.KeyRing] in the format
+	// //  `projects/*/locations/*/keyRings/*`.
+	// Name *string `json:"name,omitempty"`
+
+	// The self link of the created KeyRing in the format projects/{project}/locations/{location}/keyRings/{name}.
+	SelfLink *string `json:"selfLink,omitempty"`
```

Comment:
or `externalRef`?

Reply:
So this PR is just creating go types for the existing structure, I don't think we want to change the schema.  Renaming selfLink -> externalRef might be considered a breaking change, but regardless it likely shouldn't happen in this PR (IMO)